r/R/enums.R

```diff
@@ -60,3 +60,9 @@ StatusCode <- enum("arrow::StatusCode",
   PlasmaObjectExists = 20L, PlasmaObjectNonexistent = 21L,
   PlasmaStoreFull = 22L, PlasmaObjectAlreadySealed = 23L
 )
+
+#' @rdname DataType
+#' @export
+FileMode <- enum("arrow::io::FileMode",
+  READ = 0L, WRITE = 1L, READWRITE = 2L
+)
```

Comment:
Is there a way to get these enums from the C++ headers rather than hard-coding the values here?

Reply:
I’m not aware of a way to do that. Perhaps we can parse the headers somehow, IIRC once it’s compiled, it’s too late. @jjallaire ?

Reply:
Yes, we'd have to parse the headers during some sort of R code generation phase. We don't currently have a convenient place to do this though.

Reply:
I see. It would be interesting if Rcpp could generate constant values